### DISCUSSION TITLE
DPS5020 freeze

### DISCUSSION THREAD

**Discussion by theeye1000** [syssi/esphome-dps#25] [Q&A]
Hallo zusammen,

wenn die Werte des DPS5020 alle 10 Sekunden ändern lassen, friert dieser nach ein paar Stunden ein und ich kann keine Daten mehr senden oder empfangen. Hat noch jemand dieses Problem? Da hilft dann nur einmal Stromlos schalten. Dann funktioniert er wieder für ein paar Stunden.

Wenn ich keine Werte schreibe, sondern nur abfrage, dann passiert das nicht. 

Viele Grüße
Marco

**Comment by syssi** [comment]
Handelt es sich um ein frisches Setup oder hast du den DPS schon ein paar Jahre in Betrieb?

**Reply by theeye1000** [reply]
Der DPS ist komplett neu. Das Board ist gerade mal 1 Woche alt.

**Reply by syssi** [reply]
Und du schaltest den DPS stromlos und nicht den ESP, richtig?

**Reply by theeye1000** [reply]
Genau, ich bekomme erst wieder Werte, wenn ich das DPS neu starte. Auch das Display des DPS geht erst nach einem Neustart wieder.

**Reply by syssi** [reply]
@romeox44 Kennst du dieses Fehlerbild? Ich habe bisher noch nie davon gehoert.

**Reply by theeye1000** [reply]
Könnt Ihr denn euren DPS alle 5 Sekunden mit neuen Daten bespielen?

**Reply by romeox44** [reply]
Hallo, ich hatte mit dem DPS noch nie Probleme, es läuft noch heute wie am ersten Tag. Die Werte kommen bei mir in Echtzeit.

**Reply by theeye1000** [reply]
Die Werte können in Echtzeit abgelesen werden, ohne dass das DPS einfriert. Wenn ich jedoch alle 5 Sekunden einen Wert schreibe, friert das DPS nach einigen Stunden ein. 

Könnte es sein, dass alle 10 Sekunden zu viel ist und der DPS diese nicht mehr verarbeiten kann und einfriert? Ähnlich einem Buffer Overflow

**Reply by syssi** [reply]
Ich habe eine Änderung rückgängig gemacht, welche möglicherweise zu diesen Freezes geführt hat. Kannst du den ESP erneut flashen? Ist es nun stabil?

**Reply by theeye1000** [reply]
Ich flashe gerade den ESP und beobachte es. LG

**Reply by theeye1000** [reply]
Hallo, benutzt Ihr nicht die Originale Firmware ? Bei der Originalen Firmware habe ich immer wieder einen freeze gehabt. Zusätzlich, wenn ich den DPS einschalten oder ausschalten möchte funktioniert das nicht immer mit der Automatisierung in Homeassistant. Er nimmt die Befehle nicht immer an.

**Reply by syssi** [reply]
Ja für gewöhnlich kommt die orig. Firmware zum Einsatz aber auch die Profimax-Firmware sollte funktionieren. Wir müssen genau zwischen Freezes (DPS muss vom Strom genommen werden) und "hat einen Befehl nicht angenommen" unterscheiden. Letzteres kann in der Tat passieren, weil die Implementierung nicht mit eine Queue arbeitet und im schlimmsten Fall zwei Befehle "gleichzeitig" rausgeschickt werden.

**Reply by theeye1000** [reply]
Okay, danke. Dann bleiben wir beim Freeze. Wenn ich alle 5 Sekunden den Current anpassen will, hängt sich das System nach ein paar Stunden auf. Wie oft schreibt ihr Daten zum DPS?

Zur Info: Homeassistant liest über den ESP32 Daten aus dem DPS aus und berechnet einen Wert, der als max current über den ESP32 in den DPS geschrieben wird. Dies geschieht alle 5 Sekunden.

**Reply by syssi** [reply]
@romeox44 hat frueher alle 2 Sekunden einen neuen Wert an den DPS geschickt. Mittlerweile sendet er alle 10 Sekunden einen Wert. Ich habe die Loesung nicht produktiv im Einsatz.

**Reply by theeye1000** [reply]
Was meinst du mit, du hast die Lösung nicht Produktiv im Einsatz ?

**Reply by romeox44** [reply]
Er hat das damals für mich geschrieben. Bei mir läuft das jetzt glaube seit 2 Jahren ohne Probleme. Ich habe die originale Firmware drauf.

**Reply by syssi** [reply]
Mein DPS schlummert in einer Bastelkiste.

**Comment by romeox44** [comment]
Ja läuft ohne Probleme ;)
